### PR TITLE
:book: add github actions to push docs 

### PR DESCRIPTION
**Pull request by avinal**
## Summary

This GitHub Actions does the following:
- generates CLI docs
- generates CRD refs
- saves new tags as archive in gh-pages branch for doc versioning
- builds and deploys hugo docs to github pages

## Needs

- depends on PR #2120
- depends on PR #2219

### CODE REVIEW

File: .github/workflows/docs-gen-and-push.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: Generate and push docs
+
+on:
+  push:
+    branches:
+      - main
+      - "release-*"
```
Comment (sttts) [inline .github/workflows/docs-gen-and-push.yaml:7]: is this correct to have so many branches?

Comment (avinal) [reply .github/workflows/docs-gen-and-push.yaml:7]: My idea was to have versioned docs if we intend to do so. We can leave it to just `main` for now.

**Comment by avinal** [comment]
Need to fix `hack/verify-go-versions.sh`, it is grepping `go version` from `hugo version` 😆

**Comment by avinal** [comment]
> Need to fix `hack/verify-go-versions.sh`, it is grepping `go-version` from `hugo-version` laughing

fixed ✅

**Pull request merged**